fontbe/src/features.rs

```diff
@@ -111,22 +302,167 @@ fn write_debug_fea(context: &Context, is_error: bool, why: &str, fea_content: &s
     };
 }
 
+fn create_glyphmap(glyph_order: &IndexSet<GlyphName>) -> GlyphMap {
+    if glyph_order.is_empty() {
+        warn!("Glyph order is empty; feature compile improbable");
+    }
+    glyph_order
+        .iter()
+        .map(|n| Into::<FeaRsGlyphName>::into(n.as_str()))
+        .collect()
+}
+
+fn push_identifier(fea: &mut String, identifier: &KernParticipant) {
+    match identifier {
+        KernParticipant::Glyph(name) => fea.push_str(name.as_str()),
+        KernParticipant::Group(name) => {
+            fea.push_str("@");
+            fea.push_str(name.as_str());
+        }
+    }
+}
+
+/// Create a single variable fea describing the kerning for the entire variation space.
+///
+/// No merge baby! - [context](https://github.com/fonttools/fonttools/issues/3168#issuecomment-1608787520)
+///
+/// To match existing behavior, all kerns must have values for all locations for which any kerning is specified.
+/// See <https://github.com/fonttools/fonttools/issues/3168#issuecomment-1603631080> for more.
+/// Missing values are populated using the https://unifiedfontobject.org/versions/ufo3/kerning.plist/#kerning-value-lookup-algorithm.
+/// In future it is likely sparse kerning - blanks filled by interpolation - will be permitted.
```

Comment:
sparse is only for non-default locations, default is and must remain dense

Reply:
I don't agree that in _source_ default needs to be dense but I think I will kick all the sparse bits out of this PR so we can have further discussion.